src/main/java/org/spongepowered/api/entity/attribute/type/AttributeOperation.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.entity.attribute.type;
+
+import org.spongepowered.api.CatalogType;
+import org.spongepowered.api.entity.attribute.Attribute;
+import org.spongepowered.api.util.annotation.CatalogedBy;
+
+/**
+ * Represents an operation which can be applied to transform the value of an
+ * {@link Attribute}.
+ */
+@CatalogedBy(AttributeOperations.class)
+public interface AttributeOperation extends CatalogType {
```

Comment:
Should we expose `AttributeOperation` as a `DoubleFunction` as well? Or a `DoubleBiFunction` of sorts?

Reply:
Looking at the code again mojang seems to store all the modifiers by their operations within the EntityAttributeInstance [Attribute in API] (sorry just like using yarn names). 

It could be done but since the game seems to calculate from ADD -> MULTIPLY_BASE -> MULTIPLY_TOTAL the predictions might be different depending on which operation is being added into the modifier and where it sits within the list of Modifiers on the attribute.